### PR TITLE
`fal run` is aware of dbt --target flag

### PR DESCRIPTION
**Pull request by mkam039**
related to https://github.com/fal-ai/fal/issues/261

### CODE REVIEW

File: integration_tests/features/run_with_custom_target.feature [hunk 1]
```diff
@@ -0,0 +1,67 @@
+Feature: `run` command with a custom profile target
+  Background:
+    Given the project 007_run_with_custom_target
```
Comment (chamini2) [inline integration_tests/features/run_with_custom_target.feature:3]: How is this project different from all the others?

Comment (chamini2) [reply integration_tests/features/run_with_custom_target.feature:3]: Maybe what you can change is the `--profiles-dir profiles/broken` in calls and use a common project

File: integration_tests/features/run_with_custom_target.feature [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      dbt run --profiles-dir projects/007_run_with_custom_target --project-dir $baseDir --target custom
+      """
+    And the following command is invoked:
+      """
+      fal run --profiles-dir projects/007_run_with_custom_target --project-dir $baseDir --script fal_scripts/after.py
+      """
+    Then the following scripts are ran:
+      | agent_wait_time.after.py |
+
+  Scenario: when false script is selected, nothing runs with custom target
```
Comment (chamini2) [inline integration_tests/features/run_with_custom_target.feature:58]: I feel this does not really test the custom target but script selection.

**Review by chamini2 — COMMENTED** [review]
2 general ideas about this:

1. should we offer `--target` for fal run itself? Have more precedence for the flag than reading run results
2. I think we should offer `--target` for fal flow

**Comment by mkam039** [comment]
> 2 general ideas about this:
> 
>     1. should we offer `--target` for fal run itself? Have more precedence for the flag than reading run results
> 
>     2. I think we should offer `--target` for fal flow

@chamini2 for sure, I created a ticket for flow already and we can add --target to fal run in another ticket.

File: src/faldbt/lib.py [hunk 3]
```diff
@@ -116,23 +119,23 @@ def _get_target_relation(
...
+) -> RemoteRunResult:
+    _, result = _execute_sql(project_dir, profiles_dir, sql, target)
     return result
 
 
 def fetch_target(
-    project_dir: str,
-    profiles_dir: str,
-    target: CompileResultNode,
+    project_dir: str, profiles_dir: str, target: CompileResultNode, profile_target=None
```
Comment (chamini2) [inline src/faldbt/lib.py:130]: can we keep all the naming the same? `target` or `profile_target`?

**Review by chamini2 — APPROVED** [review]
Looks good!

**Pull request merged**